### ISSUE TITLE
Complete swagger:route generator based on full OpenAPI router spec

### ISSUE THREAD

**Issue by GlenDC** [go-swagger/go-swagger#790]
## Problem statement

Currently the `swagger:route` generator is incomplete and doesn't allow everything that's defined in the Paths Object and its children as specified in the OpenAPI (Swagger 2.0) spec.

Defining responses linked to status codes is already possible, but can currently only be defined when linked to an actual Golang type linked to an existing `swagger:response` Godoc comment. This is fine, and this behavior should remain to be supported, there are however situations where one cannot or doesn't want to link a REST API response to an explicitly defined type. In such cases it should be possible to define the response inline, as one can do in an actual OpenAPI spec file.

Defining parameters is currently also already possible, but again only when linked to an actual Golang type. Again this behavior should probably remain supported, but just as with Responses, we probably also should support inline definitions of parameters, as one would usually do when defining a pure spec file.

This is an enchantment issue to announce the work I'll be doing on this codebase to make this happen. I'll also keep a checklist of the tasks involved to get this issue 'solved', such that anyone can follow the work in progress. Please get involved if:

- You are working on a very similar issue already;
- You have feedback on the proposal and/or work in progress;
- You know of related issues that I should take a look at, and potentially fix;

## Tasks

- [ ] Support `Parameters` as a multi-line tag, as part of the existing `swagger:route` specification;
- [ ] Make `Responses` work similar to how it the [OpenAPI::ResponseObject](http://swagger.io/specification/#responseObject) works, a reference should still be possible, but defining its schema completely inline should be possible as well;
- [ ] Make it possible to define body response references, where the type does not need to be defined within an inline `Body` variable containing the actual definition;

## Remaining open questions

- What to do when a parameter is both defined inline as part of the `swagger:route` comment, and as a `swagger:parameter` comment?

## Related issues

- #782
- #779
- #361

**Comment by GlenDC** [comment]
This issue should probably be closed, as it might not be needed anymore after all.

Talking to @casualjim on slack helped me clarify things. It seems like I was making suggestions because I didn't fully understand the design behind go-swagger. I opened issue #791 so that documentation for this can be added, as I wasn't the first to talk to @casualjim about this. Secondly I opened issue #793 to support a new annotation, `operation`, that allows you to define a Path/Route as you would specify a `Path Object` in your Swagger YAML Specification.
